### ISSUE TITLE
Organize tests in test/ folder

### ISSUE THREAD

**Issue by TheZ3ro** [keepassxreboot/keepassx#56]
## Expected Behavior

All the application tests should be in the tests/ folder
Platform-dependent tests should be in test/<platform> folder

## Current Behavior

Some tests are not in the tests folder and behave feature-dependent and not platform-dependent
(like X11 Autotype)

**Comment by TheZ3ro** [comment]
In reality this is fixed.

The current behavior is fine because the **src/autotype/test** folder is a Test interface for the Qt PluginLoader (instead of Mac, X11 or Windows interface).

Can we close this? @droidmonkey

**Comment by droidmonkey** [comment]
Yes I think we can close this as I will likely do the same thing in the http folder for the http tests (mock of the Chrome/Firefox plugin interface).

**Comment by TheZ3ro** [comment]
By me is fine renaming the folder to "mock" or "stub".
